main/SQLHelper.cpp

```diff
@@ -3433,6 +3433,7 @@ uint64_t CSQLHelper::UpdateValueInt(const int HardwareID, const char* ID, const
 			GetLightStatus(devType, subType, switchtype,nValue, sValue, lstatus, llevel, bHaveDimmer, maxDimLevel, bHaveGroupCmd);
 
 			bool bIsLightSwitchOn=IsLightSwitchOn(lstatus);
+			std::string slevel = sd[6];
```

Comment:
No, it's only ever used as a string, in the search/replace. If we do update the level, we then convert that integer into a string for this.